sympy/sets/sets.py

```diff
@@ -584,7 +584,7 @@ def _eval_Eq(self, other):
         if len(self.args) != len(other.args):
             return false
 
-        return And(*map(lambda x, y: Eq(x, y), self.args, other.args))
+        return And(*[Eq(x, y) for x, y in zip(self.args, other.args)])
```

Comment:
switch `[...]` to `(...)`, no need to make a list here if we're just going to splat the args into the function. A generator expression would be better. Same with below.